File: Application/Sources/Helpers/Categories/UIColor+PlaySRG.m
//
//  Copyright (c) SRG SSR. All rights reserved.
//
//  License information is available from the LICENSE file.
//

#import "UIColor+PlaySRG.h"

@import SRGAppearance;

@implementation UIColor (PlaySRG)

+ (UIColor *)play_notificationRedColor
{
    return [UIColor srg_colorFromHexadecimalString:@"#ed3323"];
}

+ (UIColor *)play_greenColor
{
    return [UIColor srg_colorFromHexadecimalString:@"#347368"];
}

+ (UIColor *)play_orangeColor
{
    return [UIColor srg_colorFromHexadecimalString:@"#df5200"];
}

+ (UIColor *)play_popoverGrayBackgroundColor
{
    if (UIDevice.currentDevice.userInterfaceIdiom == UIUserInterfaceIdiomPad) {
        return [UIColor srg_colorFromHexadecimalString:@"#2d2d2d"];
    }
    else {
        return [UIColor srg_colorFromHexadecimalString:@"#1a1a1a"];
    }
}

+ (UIColor *)play_grayThumbnailImageViewBackgroundColor
{
    return [UIColor srg_colorFromHexadecimalString:@"#202020"];
}

+ (UIColor *)play_blackDurationLabelBackgroundColor
{
    return [UIColor colorWithWhite:0.f alpha:0.5f];
}

@end

File: Application/Sources/Helpers/Extensions/UIColor+PlaySRG.swift
//
//  Copyright (c) SRG SSR. All rights reserved.
//
//  License information is available from the LICENSE file.
//

import SRGAppearance

extension UIColor {
    @objc static var play_black80a: UIColor {
        .black.withAlphaComponent(0.8)
    }

    @objc static var play_notificationRed: UIColor {
        play_hexadecimal("#ed3323")
    }

    @objc static var play_orange: UIColor {
        play_hexadecimal("#df5200")
    }

    static var play_popoverGrayBackground: UIColor {
        if UIDevice.current.userInterfaceIdiom == .pad {
            play_hexadecimal("#2d2d2d")
        } else {
            play_hexadecimal("#1a1a1a")
        }
    }

    @objc static var play_blackDurationLabelBackground: UIColor {
        UIColor(white: 0.0, alpha: 0.5)
    }

    private static func play_hexadecimal(_ string: String) -> UIColor {
        UIColor.hexadecimal(string) ?? UIColor.white
    }
}
